### PR TITLE
[Feature] support checkpoint with subdir

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
1. support write checkpoint to subdir
2. support transfer subdir's image

## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by starrocks-xupeng** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 78 (07.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|8|00.00%|[346, 347, 642, 643, 646, 653, 657, 658]|
|:large_blue_circle:|com/starrocks/http/meta/MetaService.java|0|26|00.00%|[72, 73, 74, 75, 77, 86, 111, 112, 113, 114, 116, 120, 205, 206, 207, 208, 209, 220, 221, 229, 230, 249, 272, 273, 274, 275]|
|:large_blue_circle:|com/starrocks/master/Checkpoint.java|6|44|13.64%|[90, 91, 92, 94, 98, 117, 129, 130, 154, 180, 181, 185, 191, 194, 195, 197, 198, 199, 201, 202, 203, 204, 205, 206, 209, 211, 212, 213, 214, 216, 217, 218, 219, 220, 221, 222, 225, 226]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 78 (07.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|8|00.00%|[346, 347, 642, 643, 646, 653, 657, 658]|
|:large_blue_circle:|com/starrocks/http/meta/MetaService.java|0|26|00.00%|[72, 73, 74, 75, 77, 86, 111, 112, 113, 114, 116, 120, 205, 206, 207, 208, 209, 220, 221, 229, 230, 249, 272, 273, 274, 275]|
|:large_blue_circle:|com/starrocks/leader/Checkpoint.java|6|44|13.64%|[90, 91, 92, 94, 98, 117, 129, 130, 154, 180, 181, 185, 191, 194, 195, 197, 198, 199, 201, 202, 203, 204, 205, 206, 209, 211, 212, 213, 214, 216, 217, 218, 219, 220, 221, 222, 225, 226]|

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 1]
```diff
@@ -626,24 +637,25 @@ private boolean getVersionFileFromHelper(Pair<String, Integer> helperNode) throw
...
 
-        URL infoUrl = new URL("http://" + helperNode.first + ":" + Config.http_port + "/info");
+        URL infoUrl = new URL("http://" + helperNode.first + ":" + Config.http_port + "/info?subdir=" + subDir);
         StorageInfo info = getStorageInfo(infoUrl);
         long version = info.getImageJournalId();
         if (version > localImageVersion) {
             String url = "http://" + helperNode.first + ":" + Config.http_port
-                    + "/image?version=" + version;
-            LOG.info("start to download image.{} from {}", version, url);
+                    + "/image?version=" + version + "&subdir=" + subDir;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:651]: Better to keep this log.

File: fe/fe-core/src/main/java/com/starrocks/http/meta/MetaService.java [hunk 2]
```diff
@@ -184,22 +202,32 @@ public void executeGet(BaseRequest request, BaseResponse response) {
...
                 return;
             }
 
             String url = "http://" + machine + ":" + portStr
-                    + "/image?version=" + versionStr;
+                    + "/image?version=" + versionStr + "&subdir=" + subDirStr;
             String filename = Storage.IMAGE + "." + versionStr;
 
-            File dir = new File(GlobalStateMgr.getCurrentState().getImageDir());
+            String realDir = GlobalStateMgr.getCurrentState().getImageDir() + subDirStr;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/http/meta/MetaService.java:229]: Is `subDirStr` prefixed with `/`?

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java (hunk 1 above)
Comment (starrocks-xupeng) [reply fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:651]: fixed

**Review by gengjun-git — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 78 (07.69%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|8|00.00%|[346, 347, 642, 643, 646, 654, 658, 659]|
|:large_blue_circle:|com/starrocks/http/meta/MetaService.java|0|26|00.00%|[72, 73, 74, 75, 77, 86, 111, 112, 113, 114, 116, 120, 205, 206, 207, 208, 209, 220, 221, 229, 230, 249, 272, 273, 274, 275]|
|:large_blue_circle:|com/starrocks/leader/Checkpoint.java|6|44|13.64%|[90, 91, 92, 94, 98, 117, 129, 130, 154, 180, 181, 185, 191, 194, 195, 197, 198, 199, 201, 202, 203, 204, 205, 206, 209, 211, 212, 213, 214, 216, 217, 218, 219, 220, 221, 222, 225, 226]|

**Pull request merged**